deployer/deployer-services/src/main/java/io/hyscale/deployer/services/handler/impl/V1PodHandler.java

```diff
@@ -251,15 +253,26 @@ public InputStream tailLogs(ApiClient apiClient, String serviceName, String name
 			}
 			podName = v1Pods.get(0).getMetadata().getName();
 		}
+		OkHttpClient existingHttpClient = apiClient.getHttpClient();
 		try {
-			apiClient.getHttpClient().setReadTimeout(120, TimeUnit.MINUTES);
+		    // Update timeout
+		    apiClient.setHttpClient(getUpdatedHttpClient(existingHttpClient));
 			PodLogs logs = new PodLogs(apiClient);
 			return logs.streamNamespacedPodLog(namespace, podName, containerName, null, readLines,
 					true);
 		} catch (IOException | ApiException e) {
 			LOGGER.error("Failed to tail Pod logs for service {} in namespace {} ", serviceName, namespace);
 			throw new HyscaleException(DeployerErrorCodes.FAILED_TO_TAIL_POD, serviceName, namespace);
-		}
+		} finally {
+		    apiClient.setHttpClient(existingHttpClient);
+        }
+	}
+	
+	private OkHttpClient getUpdatedHttpClient(OkHttpClient existingHttpClient) {
+	    Builder newClientBuilder = existingHttpClient.newBuilder()
+	            .readTimeout(120, TimeUnit.MINUTES);
+	    
```

Comment:
Why is this applicable only for pods and not for any other resource ?

Reply:
This is case of tailing pod logs, hence a larger timeout, in other cases we are continuing with default timeout

Reply:
Why is this in the upgraded version of the client and not in previous versions?